source/sysinfo.js

```diff
@@ -12,15 +12,14 @@ exports.getUngitLatestVersion = () => {
 }
 
 exports.getUserHash = () => {
-  return new Bluebird((resolve) => {
-    getmac.getMac((err, addr) => {
-      if (err) {
-        winston.error("attempt to get mac addr failed, using fake mac.", err);
-        addr = "abcde";
-      }
-      resolve(md5(addr));
-    });
-  });
+  let addr;
+  try {
+    addr = getMac();
+  } catch (err) {
+    winston.error("attempt to get mac addr failed, using fake mac.", err);
+    addr = "abcde";
+  }
+  return Bluebird.resolve(md5(addr));
```

Comment:
Since the function is no longer async, would it make sense to not return a promise?